cmd/mailbox-controller/controller.go

```diff
@@ -220,33 +225,37 @@ func (ctl *mbCtl) sync(ctx context.Context, refany any) bool {
 			logger.V(3).Info("Both SyncTarget and Workspace are absent or deleting, nothing to do", "mbwsName", mbwsName)
 			return false
 		}
-		err := ctl.workspaceScopedClient.Delete(ctx, mbwsName, metav1.DeleteOptions{Preconditions: &metav1.Preconditions{UID: &workspace.UID}})
+		err := ctl.spaceMgtClientset.SpaceV1alpha1().Spaces(spacemanager.ProviderNS(ctl.spaceProvider)).Delete(ctx, mbwsName, metav1.DeleteOptions{})
 		if err == nil || k8sapierrors.IsNotFound(err) {
-			logger.V(2).Info("Deleted unwanted workspace", "mbwsName", mbwsName)
+			logger.V(2).Info("Deleted unwanted space", "mbwsName", mbwsName)
 			return false
 		}
-		logger.Error(err, "Failed to delete unwanted workspace", "mbwsName", mbwsName)
+		logger.Error(err, "Failed to delete unwanted space", "mbwsName", mbwsName)
 		return true
 	}
 	// Now we have established that the SyncTarget exists and is not being deleted
 	if workspace == nil {
-		ws := &tenancyv1alpha1.Workspace{
+		//create space for mailbox cluster
+		space := &spacev1alpha1.Space{
 			ObjectMeta: metav1.ObjectMeta{
-				Annotations: map[string]string{SyncTargetNameAnnotationKey: syncTarget.Name},
+				Annotations: map[string]string{spacemanager.SpacePathAnnotationKey: ctl.espwPath},
```

Comment:
Why remove the sync target annotation?

Reply:
Added back but note that it is set on Space object not a workspace.

Reply:
Yes.
You raise the question of what user-controlled metadata is propagated from a space to the underlying provider object.  The answer is something that should be documented.